Add endpoint for `/eth/v1/beacon/states/{state_id}/fork`

rolfyone opened the pull request:
- deprecated the previous fork endpoint `/node/fork`.

- added 503 errors to a couple of endpoints that rely on chain data.

- improve path parameter handling and error reporting.

 partially addresses #2757

Signed-off-by: Paul Harris <paul.harris@consensys.net>
## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

rolfyone committed:
Add endpoint for `/eth/v1/beacon/genesis`

- deprecated the previous genesis endpoint `/node/genesis_time`.

 partially addresses #2757

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
tagged syncing endpoint as validatorRequired also.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
implement standard `/eth/v1/beacon/states/:state_id/fork` endpoint

 - deprecated the old `/node/fork` endpoint

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
Merge remote-tracking branch 'upstream/master' into 2757-get-fork

rolfyone committed:
clean up message passing from path parameters back to the calling function - now using IllegalArgumentException.

 - checked test coverage, and improved a bit.

 - tried to clean up the ChainDataUnavailable exception, but the old api's use a catch all to return a 410 so i can't put a generic handler in for the UnavailableException to return a 503.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
push V1 endpoints up to the top of the documentation

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
spotless

Signed-off-by: Paul Harris <paul.harris@consensys.net>

Nashatyrev reviewed (approved):
LGTM 

One minor note: I would suggest using `UIn64` as `validator index` type (instead of `Integer`) to match the spec

rolfyone committed:
address review comment regarding uint64 validator indexes.

since we're not supporting them in code, I've made the parser look for and raise an exception if we exceed max_int.

also fixed a minor formatting issue in the deprecated GetGenesisTime handler.

Signed-off-by: Paul Harris <paul.harris@consensys.net>
